src/crimson/os/seastore/segment_cleaner.cc

```diff
@@ -480,6 +480,9 @@ segment_id_t SegmentCleaner::allocate_segment(
 
 void SegmentCleaner::update_journal_tail_target(journal_seq_t target)
 {
+  if (target.offset.get_addr_type() == addr_types_t::RANDOM_BLOCK) {
+    return;
```

Comment:
This isn't correct.  Make this an assert until you add support for it.  (Unless that creates a problem for testing?)